ci-operator/step-registry/cucushift/upgrade/mirror-images/by-oc-adm/cucushift-upgrade-mirror-images-by-oc-adm-commands.sh

```diff
@@ -103,9 +103,14 @@ function update_icsp(){
 # Extract oc binary which is supposed to be identical with target release
 function extract_oc(){
     echo -e "Extracting oc\n"
-    local retry=5 tmp_oc="/tmp/client-2"
+    local target_version minor_version retry=5 tmp_oc="/tmp/client-2" binary='oc'
     mkdir -p ${tmp_oc}
-    while ! (env "NO_PROXY=*" "no_proxy=*" oc adm release extract -a "${CLUSTER_PROFILE_DIR}/pull-secret" --command=oc --to=${tmp_oc} ${TARGET});
+    target_version="$(env "NO_PROXY=*" "no_proxy=*" oc adm release info "${TARGET}" --output=json | jq -r '.metadata.version')"
+    minor_version="$(echo "${target_version}" | cut -f2 -d.)"
```

Comment:
just some ideas... :)
```cut -f2 -d. <<< ${target_version}```
or natively without a subshell
```
minor_version="${target_version%.*}" # cutting out z-stream part
minor_version="${minor_version#*.}" # cutting out major
```

Reply:
not a must :) 
all in all, PR looks good to me

Reply:
Your code sounds like too complex. I copied above script from other script, looks like most people like this kind of script and we'd better keep same coding style, right?

Reply:
> we'd better keep same coding style

makes sense. even though there's plenty of room for optimization, not in your code specifically but the whole repo in general, however for simplicity sake and in terms of code familiarity I guess you're right.